libraries/shared/storage/utils/value.go

```diff
@@ -21,22 +21,36 @@ type ValueType int
 const (
 	Uint256 ValueType = iota
 	Uint48
+	Uint128
 	Bytes32
 	Address
+	PackedSlot
 )
 
 type Key string
 
 type StorageValueMetadata struct {
-	Name string
-	Keys map[Key]string
-	Type ValueType
+	Name        string
+	Keys        map[Key]string
+	Type        ValueType
+	PackedNames map[int]string    //position in map (zero indexed) => name of packed item
+	PackedTypes map[int]ValueType //position in map (zero indexed)=> type of packed item
 }
 
-func GetStorageValueMetadata(name string, keys map[Key]string, t ValueType) StorageValueMetadata {
+func GetStorageValueMetadata(name string, keys map[Key]string, valueType ValueType) StorageValueMetadata {
+	return getMetadata(name, keys, valueType, nil, nil)
+}
+
+func GetStorageValueMetadataForPackedSlot(name string, keys map[Key]string, valueType ValueType, packedNames map[int]string, packedTypes map[int]ValueType) StorageValueMetadata {
+	return getMetadata(name, keys, valueType, packedNames, packedTypes)
+}
+
+func getMetadata(name string, keys map[Key]string, t ValueType, packedNames map[int]string, packedTypes map[int]ValueType) StorageValueMetadata {
 	return StorageValueMetadata{
```

Comment:
This is probably unnecessary but I'm wondering if we might want to put a guard here to check that if `t == PackedSlot` then `packedNames != nil` and `packedTypes != nil` (and vice versa if `t != PackedSlot`)